pkg/operator/operator.go

```diff
@@ -811,6 +812,38 @@ func (op *Reporting) newPrometheusConn(promConfig promapi.Config) (prom.API, err
 	return prom.NewAPI(client), nil
 }
 
+func (op *Reporting) handleExpiredReport(report *metering.Report, now time.Time) error {
+	// check if the Report is being deleted already
+	if report.DeletionTimestamp != nil {
+		return fmt.Errorf("report was already marked for deletion")
```

Comment:
Just thinking out loud here - is this the best way to handle this kind of event, i.e. should we be returning an error here, or should we just complain about it in the logs and return nil?

Reply:
sounds good. done.